### PR TITLE
[v3] Dir to uss encoding flag

### PR DESCRIPTION
**Pull request by jace-roell**
**What It Does**
Added support for `--encoding` flag to `zowe upload dir-to-uss` to allow for encoding uploaded directories for command group consistency

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @jace-roell for the quick fix and for adding a system test!

File: packages/cli/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 All notable changes to the Zowe CLI package will be documented in this file.
 
+## Recent Changes
+
+- BugFix: Added support for `--encoding` flag to `zowe upload dir-to-uss` to allow for encoding uploaded directories for command group consitency [#2337](https://github.com/zowe/zowe-cli/issues/2337)
```
Comment (traeok) [inline packages/cli/CHANGELOG.md:7]:
```suggestion
- BugFix: Added support for `--encoding` flag to `zowe upload dir-to-uss` to allow for encoding uploaded directories for command group consistency. [#2337](https://github.com/zowe/zowe-cli/issues/2337)
```

File: packages/zosfiles/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 All notable changes to the Zowe z/OS files SDK package will be documented in this file.
 
+## Recent Changes
+
+- BugFix: Added support for `--encoding` flag to `zowe upload dir-to-uss` to allow for encoding uploaded directories for command group consitency [#2337](https://github.com/zowe/zowe-cli/issues/2337)
```
Comment (traeok) [inline packages/zosfiles/CHANGELOG.md:7]:
```suggestion
- BugFix: Added support for `--encoding` flag to `zowe upload dir-to-uss` to allow for encoding uploaded directories for command group consistency. [#2337](https://github.com/zowe/zowe-cli/issues/2337)
```

**Review by traeok — APPROVED** [review]
LGTM, thanks for adding the encoding flag! 😋

File: packages/cli/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 All notable changes to the Zowe CLI package will be documented in this file.
 
+## Recent Changes
+
+- BugFix: Added support for `--encoding` flag to `zowe upload dir-to-uss` to allow for encoding uploaded directories for command group consistency. [#2337](https://github.com/zowe/zowe-cli/issues/2337)
```
Comment (anaxceron) [inline packages/cli/CHANGELOG.md:7]: Please add articles and add "command": "Added support for **the** `bbb` flag to **the** `zzz` **command** to allow ... "

File: packages/zosfiles/CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,10 @@
 
 All notable changes to the Zowe z/OS files SDK package will be documented in this file.
 
+## Recent Changes
+
+- BugFix: Added support for `--encoding` flag to `zowe upload dir-to-uss` to allow for encoding uploaded directories for command group consistency. [#2337](https://github.com/zowe/zowe-cli/issues/2337)
```
Comment (anaxceron) [inline packages/zosfiles/CHANGELOG.md:7]: same comment as above

**Review by anaxceron — COMMENTED** [review]
Requesting minor edits to the changelog entries

File: packages/cli/src/zosfiles/upload/dtu/DirToUSSDir.definition.ts [hunk 5]
```diff
@@ -52,7 +52,8 @@ export const DirToUSSDirDefinition: ICommandDefinition = {
         UploadOptions.asciiFiles,
         UploadOptions.attributes,
         UploadOptions.maxConcurrentRequests,
-        UploadOptions.includeHidden
+        UploadOptions.includeHidden,
+        UploadOptions.encoding
     ],
```
Comment (ATorrise) [inline packages/cli/src/zosfiles/upload/dtu/DirToUSSDir.definition.ts:57]: i like to include an example of using a new option that i added that way people know exactly how they can format their command when viewing the help. if others thumbs up would you mind adding an encoding example to the examples array?
```suggestion
          {
            description: strings.EXAMPLES.EX6, //youd have to edit this file too
            options: `"local_dir" "/a/ibmuser/my_dir" --encoding "IBM-1047"`
        }
```

**Review by ATorrise — CHANGES_REQUESTED** [review]

File: packages/cli/src/zosfiles/upload/dtu/DirToUSSDir.definition.ts (hunk 5 above)
Comment (ATorrise) [reply packages/cli/src/zosfiles/upload/dtu/DirToUSSDir.definition.ts:57]: the suggested change correlates to lines 78-79 not 57-60. it was hard to indicate this on the extension i was using to create this review

**Review by gejohnston — APPROVED** [review]

**Review by ATorrise — APPROVED** [review]

**Review by awharn — APPROVED** [review]

File: packages/zosfiles/CHANGELOG.md [hunk 6]
```diff
@@ -2,6 +2,10 @@
 
 All notable changes to the Zowe z/OS files SDK package will be documented in this file.
 
+## Recent Changes
+
+- BugFix: Added support for the `--encoding` flag to the `zowe upload dir-to-uss` to allow for encoding uploaded directories for command group consistency. [#2337](https://github.com/zowe/zowe-cli/issues/2337)
```
Comment (zFernand0) [inline packages/zosfiles/CHANGELOG.md:7]: Technically there is no need to have a changelog for the zosfiles sdk in this case.
It's fine to keep it, but if you plan to remove it, we may need to use the `no-changelog` label

File: packages/zosfiles/src/methods/upload/Upload.ts [hunk 7]
```diff
@@ -876,8 +876,9 @@ export class Upload {
             }
         } else if(options.filesMap?.fileNames.indexOf(path.basename(localPath)) > -1) {
             tempOptions.binary = options.filesMap.binary;
-
+
```
Comment (zFernand0) [inline packages/zosfiles/src/methods/upload/Upload.ts:879]: trailing spaces lint warnings

image

**Review by zFernand0 — APPROVED** [review]
LGTM as-is! 
but left a couple of comments with some minor changes 😋

**Review by anaxceron — APPROVED** [review]

**Pull request merged**